### PR TITLE
Create Sign up form with Validaiton and fetch data

### PR DESCRIPTION
**Pull request by manar-abed**
relates #9

- [ ] create sign up form component
- [ ] handel front vaildation
- [ ] fetch to signup

Screenshot 2022-04-28 171653

### CODE REVIEW

**Review by Mahmoud-Ahmad2 — APPROVED** [review]

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

File: client/src/components/index.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/* eslint-disable import/prefer-default-export */
+import { Signup } from './Forms';
+
+export {
+  Signup,
+};
```
Comment (Raghad-Mezied) [inline client/src/components/index.js:6]: Don't disable `eslint` problems, fix this

File: client/src/components/Forms/signup/index.jsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+      </Button>
+      <Modal
+        title={(
+          <div className="modal-logo-img">
+            <img src={logo} alt="logo" className="logo-img" />
+          </div>
+        )}
+        visible={isModalVisible}
+        onOk={handleOk}
+        footer={null}
```
Comment (Raghad-Mezied) [inline client/src/components/Forms/signup/index.jsx:59]: You will not add the logic of modal visibility here, just add the form and handle its functionality.

File: client/src/components/Forms/signup/index.jsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        fullName,
+        email,
+        password,
+        rool,
+      });
+      setIsModalVisible(false);
+      message.success('Welcome to Nova');
+    } catch (err) {
+      message.error('Sorry, try again');
+      console.log(err);
```
Comment (Raghad-Mezied) [inline client/src/components/Forms/signup/index.jsx:34]: Remove the console, and we need to send the error that we send from the backend so the user nhow exactly whats wrong

File: client/src/components/Forms/signup/index.jsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+import logo from '../../../assets/images/logo.png';
+import './Signup.css';
+
+function Signup() {
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [fullName, setFullName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirm, setConfirm] = useState('');
+  const [rool] = useState('');
```
Comment (Raghad-Mezied) [inline client/src/components/Forms/signup/index.jsx:20]: Why do you need a role state here? in the sign-up form, it will always be `interviewee`

File: client/src/components/Forms/signup/index.jsx [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+import axios from 'axios';
+import {
+  Modal,
+  Button,
+  Form,
+  Input,
+  message,
+} from 'antd';
+import logo from '../../../assets/images/logo.png';
+import './Signup.css';
```
Comment (Raghad-Mezied) [inline client/src/components/Forms/signup/index.jsx:12]: You are already inside the signup folder so just name it style.css

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

File: client/src/components/Forms/signup/index.jsx (hunk 2 above)
Comment (manar-abed) [reply client/src/components/Forms/signup/index.jsx:59]: If I delete it it gives me an error because I am not a user at the same time, I must be in the code
image

File: client/src/components/Forms/signup/index.css [hunk 6]
```diff
@@ -0,0 +1,45 @@
+* {
+  padding: 0;
+  margin: 0;
+  box-sizing: border-box;
+}
```
Comment (Raghad-Mezied) [inline client/src/components/Forms/signup/index.css:5]: Name this file style.css so we all have the same pattern of naming files.

File: client/src/components/Forms/signup/index.jsx [hunk 7]
```diff
@@ -0,0 +1,181 @@
+/* eslint-disable no-console */
```
Comment (Raghad-Mezied) [inline client/src/components/Forms/signup/index.jsx:1]: Delete this line

File: client/src/components/Forms/signup/index.jsx [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+  message,
+} from 'antd';
+import logo from '../../../assets/images/logo.png';
+import './index.css';
+
+const { Item } = Form;
+const { Password } = Input;
+
+function Signup() {
+  // eslint-disable-next-line no-unused-vars
```
Comment (Raghad-Mezied) [inline client/src/components/Forms/signup/index.jsx:18]: Delete this line and convert the Signup to arrow function.

File: client/src/components/Forms/signup/index.jsx [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirm, setConfirm] = useState('');
+
+  const signUp = async () => {
+    try {
+      await axios.post('/api/signup', {
+        fullName,
+        email,
+        password,
```
Comment (Raghad-Mezied) [inline client/src/components/Forms/signup/index.jsx:30]: You still need to send the role and the confirm password

File: client/src/components/Forms/signup/index.jsx [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+      });
+      setIsModalVisible(false);
+      message.success('Welcome to Nova');
```
Comment (Raghad-Mezied) [inline client/src/components/Forms/signup/index.jsx:33]: There is a message sent from the back to tell the users that they need to verify their accounts so receive this message and show it.

File: dist/utils/validation/index.js [hunk 11]
```diff
@@ -0,0 +1 @@
+"use strict";
```
Comment (Raghad-Mezied) [inline dist/utils/validation/index.js:1]: Delete all dist files

File: client/src/components/Forms/signup/index.jsx [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+              //       return Promise.resolve();
+              //     }
+              // eslint-disable-next-line max-len
+              //     return Promise.reject(new Error('The two passwords that you entered do not match!'));
+              //   },
+              // }),
+              ({ getFieldValue }) => confirmPasswordHandel(getFieldValue),
+            ]}
+          >
+            <Input.Password />
```
Comment (Raghad-Mezied) [inline client/src/components/Forms/signup/index.jsx:166]: You already destructure the Password so use it directly

File: client/src/components/Forms/signup/index.jsx [hunk 13]
```diff
@@ -0,0 +1,181 @@
...
+              },
+              // ({ getFieldValue }) => ({
+              //   validator(_, value) {
+              //     if (!value || getFieldValue('password') === value) {
+              //       return Promise.resolve();
+              //     }
+              // eslint-disable-next-line max-len
+              //     return Promise.reject(new Error('The two passwords that you entered do not match!'));
+              //   },
+              // }),
```
Comment (Raghad-Mezied) [inline client/src/components/Forms/signup/index.jsx:162]: Delete commented lines

File: client/src/components/Forms/signup/index.jsx [hunk 14]
```diff
@@ -0,0 +1,181 @@
...
+  });
+
+  return (
+    <>
+      <Button type="primary" onClick={showModal}>
+        Open Modal
+      </Button>
+      <Modal
+        title={(
+          <div className="modal-logo-img">
```
Comment (Raghad-Mezied) [inline client/src/components/Forms/signup/index.jsx:61]: You need to split these into two components, one for the btn with the shown logic and one for the modal itself with the signup logic.

File: client/src/components/Forms/signup/index.jsx [hunk 15]
```diff
@@ -0,0 +1,181 @@
...
+        return Promise.resolve();
+      }
+      return Promise.reject(new Error('The two passwords that you entered do not match!'));
+    },
+  });
+
+  return (
+    <>
+      <Button type="primary" onClick={showModal}>
+        Open Modal
```
Comment (Raghad-Mezied) [inline client/src/components/Forms/signup/index.jsx:57]: Will this message appear to the user?

File: client/src/components/Forms/signup/index.jsx [hunk 16]
```diff
@@ -0,0 +1,181 @@
...
+        fullName,
+        email,
+        password,
+      });
+      setIsModalVisible(false);
+      message.success('Welcome to Nova');
+    } catch (err) {
+      message.error(`Sorry, try again , ${err.message}`);
+    }
+  };
```
Comment (Raghad-Mezied) [inline client/src/components/Forms/signup/index.jsx:37]: Try this and make sure if its work

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

File: client/src/redux/features/auth/authSlice.js [hunk 17]
```diff
@@ -16,9 +16,9 @@ const authSlice = createSlice({
   name: 'auth',
   initialState,
   reducers: {
-    setUser: (state, action) => {
-      state.user = action.payload;
```
Comment (Raghad-Mezied) [inline client/src/redux/features/auth/authSlice.js:20]: Why did you delete this reducer?
And remove all the redux logic in sign up since the user will not be verified.

File: client/src/components/Login/LoginForm.jsx [hunk 18]
```diff
@@ -87,15 +87,9 @@ const LoginForm = () => {
...
-        <Button type="primary" htmlType="submit">
+        <Button type="primary" htmlType="submit" className="signup">
           Login
         </Button>
-        <div className="have-account">
-          Don`t have an account?
-          {/* Here We need to change the anchor tag when connenct all pages together */}
-          <a href="/signup">Sign up</a>
-
-        </div>
```
Comment (Raghad-Mezied) [inline client/src/components/Login/LoginForm.jsx:98]: Why did you delete this

File: client/src/components/Login/LoginForm.jsx [hunk 19]
```diff
@@ -25,7 +25,7 @@ const LoginForm = () => {
   const onFinish = async () => {
     try {
       const { data } = await axios.post('/api/login', { email, password });
-      dispatch(setUser(data.data.user));
+      dispatch(signUpAction(data.data.user));
```
Comment (Raghad-Mezied) [inline client/src/components/Login/LoginForm.jsx:28]: Name it back setUser and destructure the user

File: client/src/components/Forms/signup/index.jsx [hunk 20]
```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react';
+import {
+  Button,
+  Modal,
+} from 'antd';
+import logo from '../../../assets/images/logo.png';
+import SignupForm from './Form';
+import './signup.css';
+
+const SignupModal = () => {
```
Comment (Raghad-Mezied) [inline client/src/components/Forms/signup/index.jsx:10]: Name this signupButton and the file name too.

File: client/src/components/Forms/signup/Form.jsx [hunk 21]
```diff
@@ -0,0 +1,149 @@
...
+          ({ getFieldValue }) => confirmPasswordHandel(getFieldValue),
+        ]}
+      >
+        <Password />
+      </Item>
+      <Item>
+        <div className="form-group-btn">
+          <Button type="primary" htmlType="submit" className="signup">
+            Register
+          </Button>
```
Comment (Raghad-Mezied) [inline client/src/components/Forms/signup/Form.jsx:142]: Add as in the login
```js
        <div className="have-account">
          Don`t have an account?
          {/* Here We need to change the anchor tag when connenct all pages together */}
          <a href="/signup">Sign up</a>

        </div>
```
In case the user has an account

File: client/src/components/Forms/signup/Form.jsx [hunk 22]
```diff
@@ -0,0 +1,149 @@
...
+    try {
+      const { data } = await axios.post('/api/signup', {
+        fullName,
+        email,
+        password,
+        confirm,
+        role: 'interviewee',
+      });
+      dispatch(signUpAction(data));
+      message.success(`Welcome ${fullName}, Signup success`);
```
Comment (Raghad-Mezied) [inline client/src/components/Forms/signup/Form.jsx:39]: Receive the message from the back and show it, so we tell the users that they need to verify their accounts.

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by Raghad-Mezied — APPROVED** [review]
